### DISCUSSION TITLE
How to get the current state of the interpreter?

### DISCUSSION THREAD

**Discussion by RemyMachado** [statelyai/xstate#3650] [Q&A] [answered]
It's probably really easy, but I couldn't find an answer.

```typescript
// 1) I create the machine
const fsm = createMachine(config)

// 2) I interpret the machine
const service = interpret(fsm).onTransition((state) => {
    console.log(state.value)
})

// 3) I start the service
service.start()

// 4) send multple events
// ...

// 5) retrieve the current state
```
Now, how can I retrieve the service current state at **5)** ? It's the `fsm.initialState` since I didn't send any event yet, but let's imagine I sent events **without assigning the result of `service.send()`**.

I'm looking for a property like `service.currentState`.

I know there is `service.nextState`, but I don't want the next, I want the current one.

Thank you for your help.

**Comment by Andarist** [comment, marked as answer]
With `xstate` you can call `service.getSnapshot()`.

With `@xstate/fsm` you can use the `service.state` getter.
